nipype/interfaces/fsl/preprocess.py

```diff
@@ -315,6 +315,15 @@ def _format_arg(self, name, spec, value):
             formated = "-S %d %s" % (len(value), formated)
         return formated
 
+    def _parse_inputs(self, skip=None):
+        ''' ensures out_basename (argstring -o) always exists, otherwise fast
+        puts the outputs in the same folder as the inputs '''
```

Comment:
So if I understand you correctly, this is the expected behavior of `fast`, so the FAST interface should follow this behavior. Then the problem is in `_list_outputs()`, which doesn't understand that behavior (and therefore the interface fails due to trait errors when it is run with `.run()`)